task2.py

```diff
@@ -0,0 +1,115 @@
+from operator import itemgetter
+import itertools
+
+#1)Дан массив из словарей 
+data = [
+    {'name': 'Viktor', 'city': 'Kiev', 'age': 30 },
+    {'name': 'Maksim', 'city': 'Dnepr', 'age': 20},
+    {'name': 'Vladimir', 'city': 'Lviv', 'age': 32},
+    {'name': 'Andrey', 'city': 'Kiev', 'age': 34},
+    {'name': 'Artem', 'city': 'Dnepr', 'age': 50},
+    {'name': 'Dmitriy', 'city': 'Lviv', 'age': 21}
+]
+
+#1.1) отсортировать массив из словарей по значению ключа ‘age' 
+
+
+def sort_by_key(dictionary, key):
+    return sorted(dictionary, key=lambda k: k[key])
+     
+
+#1.2) сгруппировать данные по значению ключа 'city' 
+# вывод должен быть такого вида :
+
+result = {
+   'Kiev': [{'name': 'Viktor', 'age': 30 },
+            {'name': 'Andrey', 'age': 34}],
+   'Dnepr': [ {'name': 'Maksim', 'age': 20 },
+              {'name': 'Artem', 'age': 50}],
+   'Lviv': [ {'name': 'Vladimir', 'age': 32 },
+             {'name': 'Dmitriy', 'age': 21}]
+}
+
+
+def group_by_city(data):
+    result = {}
+    for i in data:
+        new_list = [{'name': i['name'], 'age': i['age']}]
+        if i['city'] not in result:
+            result[i['city']] = new_list
+            i.pop('city')
+        else:
+            result[i['city']].append(i)
+            i.pop('city')
+    return result
+
+
+# =======================================================
+# 2) У вас есть последовательность строк. Необходимо определить наиболее часто встречающуюся строку в последовательности.
+# Например:
+
+
+def most_frequent(list):
```

Comment:
nice, but don't use name `list`, because `list()` is the data structure but not variable.